src/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.tsx

```diff
@@ -131,38 +176,35 @@ const AddProfessionalCategoryModal: FC<AddProfessionalCategoryModalProps> = ({
         )}
       />
       <Box sx={styles.formWrapper}>
-        <AppAutoComplete
+        <AsyncAutocomplete
           fullWidth
+          getOptionDisabled={handleBlockOption}
+          labelField='name'
           onChange={handleMainStudyCategoryChange}
-          options={['Language1', 'Language2', 'Language3']} // @TODO: replace with actual languages from backend
+          service={categoryService.getCategoriesNames}
           textFieldProps={{
             label: `${t(
               'editProfilePage.profile.professionalTab.mainStudyCategory'
             )}*`,
-            error: Boolean(errors.mainStudyCategory),
-            helperText: errors.mainStudyCategory
+            error: Boolean(errors.category),
+            helperText: errors.category
           }}
-          value={data.mainStudyCategory}
+          value={data.category._id}
+          valueField='_id'
         />
         {data.subjects.map((subject, index) => (
```

Comment:
Move this data mapping to a separate variable

Reply:
Thank you fixed!